src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Changed.cs

```diff
@@ -93,4 +93,69 @@ public static void FileSystemWatcher_ChangeWatchedFolder()
             Utility.ExpectEvent(eventOccured, "changed");
         }
     }
+
+    [Fact]
+    public static void FileSystemWatcher_Changed_NestedDirectories()
+    {
+        using (var dir = Utility.CreateTestDirectory())
+        using (var watcher = new FileSystemWatcher())
+        {
+            AutoResetEvent eventOccured = Utility.WatchForEvents(watcher, WatcherChangeTypes.Changed);
+            watcher.Path = Path.GetFullPath(dir.Path);
+            watcher.Filter = "*";
+            watcher.IncludeSubdirectories = true;
+            watcher.EnableRaisingEvents = true;
+
+            using (var firstDir = new TemporaryTestDirectory(Path.Combine(dir.Path, "dir1")))
+            {
+                // We need to sleep to allow Linux FSW to get the event and get a WatchDescriptor on the directory
+                System.Threading.Thread.Sleep(1000);
+
+                using (var nestedDir = new TemporaryTestDirectory(Path.Combine(firstDir.Path, "nested")))
+                {
+                    // We need to sleep to allow Linux FSW to get the event and get a WatchDescriptor on the directory
+                    System.Threading.Thread.Sleep(1000);
+
+                    Directory.SetLastAccessTime(firstDir.Path, DateTime.Now);
+                    Utility.ExpectEvent(eventOccured, "changed");
+
+                    Directory.SetLastAccessTime(nestedDir.Path, DateTime.Now);
+                    Utility.ExpectEvent(eventOccured, "changed");
+                }
+            }
+        }
+    }
+
+    [Fact]
+    public static void FileSystemWatcher_Changed_FileInNestedDirectory()
+    {
+        using (var dir = Utility.CreateTestDirectory())
+        using (var watcher = new FileSystemWatcher())
+        {
+            AutoResetEvent eventOccured = Utility.WatchForEvents(watcher, WatcherChangeTypes.Changed);
+            watcher.Path = Path.GetFullPath(dir.Path);
+            watcher.Filter = "*";
+            watcher.IncludeSubdirectories = true;
+            watcher.EnableRaisingEvents = true;
+
+            using (var firstDir = new TemporaryTestDirectory(Path.Combine(dir.Path, "dir1")))
+            {
+                // We need to sleep to allow Linux FSW to get the event and get a WatchDescriptor on the directory
+                System.Threading.Thread.Sleep(1000);
+                using (var nestedDir = new TemporaryTestDirectory(Path.Combine(firstDir.Path, "nested")))
+                {
+                    // We need to sleep to allow Linux FSW to get the event and get a WatchDescriptor on the directory
+                    System.Threading.Thread.Sleep(1000);
+                    using (var nestedFile = new TemporaryTestFile(Path.Combine(nestedDir.Path, "nestedFile")))
```

Comment:
If we're able to follow my suggestion in the previous unit test, then we probably don't even need a second test here, since we're verifying the events are being received as we run the previous test, in which case this nestedFile could just be added to the innermost directory in that test.